app/models/piece.rb

```diff
@@ -167,19 +179,38 @@ def in_check?(king)
   end
   
   def not_moved_to_different_space?(x,y)
-    position_x == x && y_position == y
+    position_x == x && position_y == y
   end
   
   def is_my_turn?
     #write logic to check for whether it's the piece owner's turn here!
-    #maybe something like... @game.turn == current_player.color???? with the turn defined by color elsewhere?
+    #maybe something like... @game.turcann == current_player.color???? with the turn defined by color elsewhere?
   end
   
-  def valid_move?(x,y)
-    #i'm not sure I'm implementing in check correctly here. second set of eyes would be appreciated!
-    obstructed?(x,y) && off_the_board?(x,y) && in_check?(king) && not_moved_to_different_space?(x,y) && is_my_turn?
-    
+  def occupied?(x,y)
+	  (piece_present_at?(x,y)) && is_same_color?(x,y)
+	end
+  
+  def valid_move?(x,y,king=nil)
+    return false if obstructed?(x,y) 
+    return false if off_the_board?(x,y)
+    #return false if in_check?(king)
```

Comment:
Once again @chris-campbell / @thablabber and me got in_check working... hopefully he saved those changes somewhere

Reply:
Excellent. I'll leave it uncommented in this pr just to keep things working properly, and then when check gets merged in we should be good.